api/lib/domain/usecases/compute-campaign-analysis.js

```diff
@@ -0,0 +1,52 @@
+const _ = require('lodash');
+const CampaignAnalysis = require('../models/CampaignAnalysis');
+const CampaignTubeRecommendation = require('../models/CampaignTubeRecommendation');
+const { UserNotAuthorizedToAccessEntity } = require('../errors');
+
+module.exports = async function computeCampaignAnalysis(
+  {
+    userId,
+    campaignId,
+    campaignRepository,
+    competenceRepository,
+    targetProfileRepository,
+    tubeRepository,
+  } = {}) {
+
+  const hasUserAccessToResult = await campaignRepository.checkIfUserOrganizationHasAccessToCampaign(campaignId, userId);
+
+  if (!hasUserAccessToResult) {
+    throw new UserNotAuthorizedToAccessEntity('User does not have access to this campaign');
+  }
+
+  const competences = await competenceRepository.list();
+  const tubes = await tubeRepository.list();
+
+  const targetProfile = await targetProfileRepository.getByCampaignId(campaignId);
+  const { targetedSkillsByTubeId } = _groupByColumn(targetProfile.skills, 'tubeId');
+
+  const campaignTubeRecommendations = _forgecampaignTubeRecommendations(campaignId, tubes, competences, targetedSkillsByTubeId);
+
+  return new CampaignAnalysis({ id: campaignId, campaignTubeRecommendations });
+};
+
+function _groupByColumn(targetedSkills, columnName) {
+  const targetedSkillsByTubeId = _.groupBy(targetedSkills, columnName);
+  return { targetedSkillsByTubeId };
+}
+
+function _forgecampaignTubeRecommendations(campaignId, tubes, competences, targetedSkillsByTubeId) {
```

Comment:
```suggestion
function _forgeCampaignTubeRecommendations(campaignId, tubes, competences, targetedSkillsByTubeId) {
```
Perso je préfère `compute` à `forge` que je n'ai pas vu ailleurs dans le code.

Reply:
`forge` est utilisé aussi avec la méthode `_forgeCampaignCompetenceCollectiveResults`, je n'ai pas d'avis fort sur la question, donc je suis ok pour changer vers `compute` :)